install-dev/upgrade/php/ps_1780_add_feature_flag_tab.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+function ps_1780_add_feature_flag_tab()
+{
+    $className = 'AdminFeatureFlag';
+
+    $result = Db::getInstance()->executeS(
+        'SELECT id_tab FROM `'._DB_PREFIX_.'tab` WHERE `class_name` = \'AdminAdvancedParameters\''
+    );
+
+    if (empty($result)) {
+        return;
+    }
+    $advancedParametersTabId = (int) $result[0]['id_tab'];
```

Comment:
Please check the existence

Reply:
I thought about it but the `$result` structure is mapped to the SQL schema. Is it worth it to verify that "MySQL correctly returned the structure we have defined for it"? I am not sure it's a good thing to validate that our own SQL schema is correct... unless the developer of the shop has modified its SQL schema, but at this moment there's very few we can do 😅

Reply:
Since you checked `if (empty($result)) {` it should be good good indeed But if you want to make Franck happy you can replace it by `if (empty($result[0]['id_tab'])) {` 😅

Reply:
Never trust array :p

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/23703/commits/57b8683b8daebe4772660426b297115eb85bd502